docs/design/resource-states.md

```diff
@@ -0,0 +1,222 @@
+  ConditionSeverityNone ConditionSeverity = ""
+)
+
+// Condition defines an extension to status (an observation) of an ASO resource
+type Condition struct {
+   // Type of condition.
+   // +required
+   Type string `json:"type"`
+
+   // Status of the condition, one of True, False, or Unknown.
+   // +required
+   Status corev1.ConditionStatus `json:"status"`
+
+   // Severity with which to treat failures of this type of condition.
+   // For conditions which have positive polarity (Status == True is their normal/healthy state), this will be omitted when Status == True
+   // For conditions which have negative polarity (Status == False is their normal/healthy state), this will be omitted when Status == False.
+   // +optional
+   Severity ConditionSeverity `json:"severity,omitempty"`
+
+   // LastTransitionTime is the last time the condition transitioned from one status to another.
+   // +required
+   LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+   // The reason for the condition's last transition.
+   // Reasons should be CamelCase with no spaces. A reason is always provided, this field will not be empty
+   // +required
+   Reason string `json:"reason"`
+
+   // A human readable message indicating details about the transition. This field may be empty
+   // +optional
+   Message string `json:"message,omitempty"`
+}
+```
+
+### The `Ready` condition
+The above discusses support for arbitrary conditions. In practice at least for now, ASO will only expose a single `Ready` condition.
+The addition of a `Severity` field inspired by [Cluster API](https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20200506-conditions.md) allows different combinations of `Severity` and `Reason` to combine together and describe complex scenarios.
+
+#### The `Reason` field
+Inside of the `Ready` condition we always include a `Reason` which provides a programatically consumable reason that the resource is in the given state. 
+`Reason` is derived from multiple sources depending on where in the resource lifecycle we are. The value in `Reason` may be set by the operator itself, or be set to the result of an error or response code received from Azure. 
+
+The operator defines the following reasons:
+
+| Reason                 | Severity                        |   Meaning                                                                                                                                                                 |
+|------------------------|---------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Reconciling            | Info                            | A request has been submitted to Azure. No error has been received from Azure, but one may occur in the future.                                                            |
```

Comment:
Maybe need to be explicit that we may be waiting to find out whether it worked or not.

Reply:
Rephrased this to say that